code/game/machinery/telecomms/presets.dm

```diff
@@ -282,6 +292,72 @@ GLOBAL_LIST_EMPTY(all_static_telecomms_towers)
 	else
 		update_icon()
 
+/obj/structure/machinery/telecomms/relay/preset/tower/mapcomms/proc/handle_xeno_acquisition(turf/weeded_turf)
+	SIGNAL_HANDLER
+
+	if(corrupted)
+		return
+
+	if(!weeded_turf.weeds)
+		return
+
+	if(weeded_turf.weeds.weed_strength < WEED_LEVEL_HIVE)
+		return
+
+	if(!weeded_turf.weeds.parent)
+		return
+
+	if(!istypestrict(weeded_turf.weeds.parent, /obj/effect/alien/weeds/node/pylon/cluster))
+		return
+
+	if(ROUND_TIME < XENO_COMM_ACQUISITION_TIME)
+		addtimer(CALLBACK(src, PROC_REF(handle_xeno_acquisition), weeded_turf), (XENO_COMM_ACQUISITION_TIME - ROUND_TIME))
+		return
+
+	var/obj/effect/alien/weeds/node/pylon/cluster/parent_node = weeded_turf.weeds.parent
+
+	var/obj/effect/alien/resin/special/cluster/cluster_parent = parent_node.resin_parent
+
+	var/list/held_children_weeds = parent_node.children
+	var/cluster_loc = cluster_parent.loc
+	var/linked_hive = cluster_parent.linked_hive
+
+	parent_node.children = list()
+
+	qdel(cluster_parent)
```

Comment:
cluster_parent here can be a /obj/effect/alien/resin/special/pylon/core. Maybe you intend to delete `weeded_turf.weeds.parent` instead of `weeded_turf.weeds.parent.resin_parent`

Reply:
Errr I don't think so, this should delete the cluster itself rather than the node attached to the cluster. I fixed the hive core being eaten issue (I think)

Reply:
I am currently unable to reproduce that issue.